### PR TITLE
:seedling: Move GetMachineDeploymentNameForCluster to usage site

### PR DESCRIPTION
**Pull request by killianmuldoon**
Move this function closer to where its called, delete the package that formerly held it and change usage to internal only.

Follow up from #2383

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Might need to rebase this one when #2385 merges.

**Review by chrischdi — COMMENTED** [review]
/lgtm

Regarding the change, although I don't really get the real functionality behind the hardcoded suffix.

File: pkg/services/vmoperator/vmopmachine.go [hunk 1]
```diff
@@ -552,3 +551,9 @@ func getTopologyLabels(supervisorMachineCtx *vmware.SupervisorMachineContext) ma
 	}
 	return nil
 }
+
+// getMachineDeploymentName returns the MachineDeployment name for a Cluster.
+// This is also the name used by VSphereMachineTemplate and KubeadmConfigTemplate.
+func getMachineDeploymentNameForCluster(cluster *clusterv1.Cluster) string {
+	return fmt.Sprintf("%s-workers-0", cluster.Name)
```
[the remark this answers is not in the document]
Comment (killianmuldoon) [reply pkg/services/vmoperator/vmopmachine.go:558]: Also to make sure removing a function like this from the go API isn't an issue.

**Comment by chrischdi** [comment]
I think the func should be fine to get moved. vmoperatore should not import capv, otherwise they may get crazy circular dependencies?!

Let's just not cherry-pick it back.

Should we open an issue for the hardcoded `-workers-0-` and merge this?

/lgtm

**Comment by killianmuldoon** [comment]
/hold

For discussion at community meeting. I'd like to bring it up at the meeting to get a sense from folks about whether we should touch this stuff at all - not sure where else public functions like this might be imported. We can also talk about the hardcoded suffix and raise an issue based on discussion.

File: pkg/services/vmoperator/vmopmachine.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (randomvariable) [reply pkg/services/vmoperator/vmopmachine.go:565]: i'll need to defer to @sidharthsurana and Neha (who I can't find right now).

**Comment by killianmuldoon** [comment]
/hold cancel

Let's move forward with this one and revert if there's any issue.

**Pull request merged**